homework-selects/sql_questions.sql

```diff
@@ -1,57 +1,579 @@
+
+SELECT E.*
+FROM employees E
+    JOIN employees M
+        ON E.manager_id = M.employee_id
+    JOIN departments D
+        ON M.department_id = D.department_id
+    JOIN locations L
+        ON L.location_id = D.location_id
+    WHERE L.country_id = 'US';
+	
 --37. the names of all employees whose salary is greater than 50% of their department’s total salary bill.
+
+SELECT first_name || ' ' || last_name AS "FULL NAME"
+FROM employees
+    WHERE salary > (
+        SELECT AVG(salary)
+        FROM employees
+        WHERE department_id = department_id
+    );
+
 --38. the employee id, name ( first name and last name ), salary, department name and city for all
 --the employees who gets the salary as the salary earn by the employee which is maximum within the joining person January 1st, 2002 and December 31st, 2003.  
+
+SELECT employee_id, 
+    first_name || ' ' || last_name AS "FULL NAME",
+    salary,
+    department_name,
+    city
+FROM employees
+    JOIN departments USING (department_id)
+    JOIN locations USING (location_id)
+    WHERE salary IN (
+        SELECT MAX(salary)
+        FROM employees
+            WHERE hire_date BETWEEN TO_DATE('01/01/2002', 'dd/mm/yyyy') AND
+                        TO_DATE('31/12/2003', 'dd/mm/yyyy')
+    )
+
 --39. the first and last name, salary, and department ID for all those employees who earn more than the average salary and arrange the list in descending order on salary.
+
+SELECT first_name, 
+    last_name,
+    salary,
+    department_id
+FROM employees
+    WHERE salary > (
+        SELECT AVG(salary)
+        FROM employees
+    )
+    ORDER BY salary DESC
+	
 --40. the first and last name, salary, and department ID for those employees who earn more than the maximum salary of a department which ID is 40.
+
+SELECT first_name,
+    last_name,
+    salary,
+    department_id
+FROM employees
+    WHERE salary > (
+        SELECT MAX(salary)
+        FROM employees
+        WHERE department_id = 40
+    );
+	
 --41. the department name and Id for all departments where they located, that Id is equal to the Id for the location where department number 30 is located.
+
+SELECT department_name, 
+    department_id
+FROM departments
+    WHERE location_id = (
+        SELECT location_id
+        FROM departments 
+            WHERE department_id = 30
+    )
+	
 --42. the first and last name, salary, and department ID for all those employees who work in that department where the employee works who hold the ID 201.
+
+SELECT first_name,
+    last_name,
+    salary,
+    department_id
+FROM employees
+    WHERE department_id = (
+        SELECT department_id
+        FROM employees
+            WHERE employee_id = 201
+    );
+	
 --43. the first and last name, salary, and department ID for those employees whose salary is equal to the salary of the employee who works in that department which ID is 40.
+
+SELECT first_name,
```

Comment:
wrong query. reread '**the salary of the employee** who **works** in that **department which ID is 40**'